rust/src/server.rs

```diff
@@ -0,0 +1,736 @@
+use std::collections::BTreeMap;
```

Comment:
Sorry, review on this file is still in progress but saving my state here so you can see for now :)